jwql/utils/utils.py

```diff
@@ -368,33 +377,49 @@ def filename_parser(filename):
         r"_gs-(?P<guider_mode>(id|acq1|acq2|track|fg))" \
         r"_((?P<date_time>\d{13})|(?P<guide_star_attempt_id>\d{1}))"
 
+    # Segment guider filenames
+    # e.g. "jw00729011001_gs-id_1_image_cal.fits" or
+    # "jw00799003001_gs-acq1_2019154181705_stream.fits"
+    guider_segment = \
+        r"jw" \
+        r"(?P<program_id>\d{5})" \
+        r"(?P<observation>\d{3})" \
+        r"(?P<visit>\d{3})" \
+        r"_gs-(?P<guider_mode>(id|acq1|acq2|track|fg))" \
+        r"_((?P<date_time>\d{13})|(?P<guide_star_attempt_id>\d{1}))" \
+        r"-seg(?P<segment>\d{3})"
+
     # Build list of filename types
     filename_types = [
         stage_1_and_2,
         stage_2c,
+        stage_2_msa,
         stage_3_target_id,
         stage_3_source_id,
         stage_3_target_id_epoch,
         stage_3_source_id_epoch,
         time_series,
-        guider]
+        guider,
+        guider_segment]
 
     filename_type_names = [
         'stage_1_and_2',
         'stage_2c',
+        'stage_2_msa',
         'stage_3_target_id',
         'stage_3_source_id',
         'stage_3_target_id_epoch',
         'stage_3_source_id_epoch',
         'time_series',
-        'guider'
+        'guider',
+        'guider_segment'
     ]
 
     # Try to parse the filename
     for filename_type, filename_type_name in zip(filename_types, filename_type_names):
 
-        # If full filename, try using suffix
-        if not file_root_name:
+        # If full filename, try using suffix, except for *msa.fits files
+        if not file_root_name and 'msa.fits' not in filename:
```

Comment:
Is the 'msa.fits' the only suffix we will ever need isolated here?  Is this the only location 'msa.fits' is isolated in logic?  If there is potential for more in the future, this would be another good spot to put a variable list of excluded suffixes for this situation

Reply:
I also put this in constants.py. Not sure if there will be any others, but there is now a place to add them later.